### PR TITLE
Add support for 8x16 kernels in TFLM.

### PR DESCRIPTION
**Pull request by njeffrie**
Add support for 8x16 kernels in TFLM.

Support 16-bit activations and 8-bit weights for a subset of the TFLM kernels.

BUG=http://b/184839019

### CODE REVIEW

File: tensorflow/lite/micro/kernels/sub.cc [hunk 1]
```diff
@@ -187,7 +196,29 @@ TfLiteStatus EvalSubQuantized(TfLiteContext* context, TfLiteNode* node,
...
+            op_params, tflite::micro::GetTensorShape(input1),
+            tflite::micro::GetTensorData<int16_t>(input1),
+            tflite::micro::GetTensorShape(input2),
+            tflite::micro::GetTensorData<int16_t>(input2),
+            tflite::micro::GetTensorShape(output),
+            tflite::micro::GetTensorData<int16_t>(output), false);
+      }
+      break;
+    }
+    case kTfLiteUInt8: {
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/sub.cc:221]: can we do a follow-on PR to remove uint8 support?

File: tensorflow/lite/micro/micro_utils.h [hunk 2]
```diff
@@ -44,11 +44,11 @@ T FloatToQuantizedType(const float value, const float scale, int zero_point) {
 
 template <typename T>
 T FloatToSymmetricQuantizedType(const float value, const float scale) {
-  int32_t result = round(value / scale);
-  result =
-      std::max(static_cast<int32_t>(std::numeric_limits<T>::min() + 1), result);
-  result =
-      std::min(static_cast<int32_t>(std::numeric_limits<T>::max()), result);
+  std::int64_t result = round(value / scale);
```
Comment (advaitjain) [inline tensorflow/lite/micro/micro_utils.h:47]: why do we need to change to 64 bits? might be worth a comment.

**Review by advaitjain — CHANGES_REQUESTED** [review]
nit comments only

File: tensorflow/lite/micro/kernels/sub.cc (hunk 1 above)
Comment (njeffrie) [reply tensorflow/lite/micro/kernels/sub.cc:221]: Ack, will do.

File: tensorflow/lite/micro/micro_utils.h (hunk 2 above)
Comment (njeffrie) [reply tensorflow/lite/micro/micro_utils.h:48]: Added a comment explaining that 8x16 conv requires a 64-bit bias, so numeric_limits<int64_t> must be an int64.

**Review by advaitjain — APPROVED** [review]

**Pull request merged**